### ISSUE TITLE
perform permanent redirects

### ISSUE THREAD

**Issue by Nowaker** [virtkick/http-master#76]
I think this should be the default. Redirects directly in HTTP server's configuration file are things that usually stay there for a long time. All the redirects I have are 100% permanent - they redirect from non-SSL to SSL, or from non-www to www. That said, permanent redirects should be the default. 

An option could be introduced to change this behavior. (But I'm not that convinced, if asking me) For example:

```
redirect: {
  "url": "target"
},
tempRedirect: {
  "url": "target"
}
```

**Comment by szarouski** [comment]
I think that 301 should be default too. Is there a way to change that with existing configuration options?

**Comment by Rush** [comment]
You can edit `modules/middleware/redirect.js` and edit line 11. I am little bit concerned about caching issues to change it to 301 by default.

**Comment by danielmahon** [comment]
I agree, this should be configurable, but I wonder what the current browser methods are to cache/uncache 301s? Suposively you're able to set `Cache Control` header on 301s, so if you set `no-cache` will that clear it? I am currently using http-master as a http(s) non-www to https www proxy and I want these redirects to be permanent to increase speed, avoid multiple redirects, but  hate making anything "permanent". Is it better to just use a cached 302?
